gui

```diff
@@ -1,28 +1,280 @@
 #!/bin/bash
 
-set -a #export all functions and variables for child processes like runonce()
-
-DIRECTORY="$(readlink -f "$(dirname "$0")")"
+if ! command -v yad &>/dev/null; then
+  error "YAD needs to be installed to run Pi-Apps."
```

Comment:
the api functions haven't been sourced yet, if you pass this check, `error` is not available.

its probably best you just replace this with the appropriate echo and exit from the error function
```suggestion
  echo -e "\e[91mYAD needs to be installed to run Pi-Apps.\e[0m" 1>&2
  exit 1
```